apps/node/server.js

```diff
@@ -0,0 +1,139 @@
+const express = require("express");
+const axios = require("axios");
+const dotenv = require("dotenv").config();
+const crypto = require("crypto");
+const qs = require("qs");
+const { getState, setState } = require("./dal");
+
+const ATTENTIVE_API_URL = "https://api.attentivemobile.com/v1";
+const ACCESS_TOKEN_ENDPOINT =
+  "https://api.attentivemobile.com/v1/authorization-codes/tokens";
+
+const CLIENT_ID = dotenv.parsed.CLIENT_ID; // Your client id
+const CLIENT_SECRET = dotenv.parsed.CLIENT_SECRET; // Your secret
+const REDIRECT_URI = dotenv.parsed.REDIRECT_URI; // Your redirect uri
+const PORT = 3002;
+
+let data = {};
+
+let app = express();
+app.use(express.json());
+
+app.get("/", (req, res) => {
+  res.send({ message: "Your Application is running successfully!" });
+});
+
+app.get("/install", async (req, res) => {
+  // Generates a random string containing numbers and letters
+  const state = generateRandomString(10);
+
+  // List of scopes that your app needs
+  const scopes = ["ecommerce:write", "events:write", "subscriptions:write"];
+
+  setState(state, {});
+  const redirect_url =
+    `https://ui.attentivemobile.com/integrations/oauth-install?client_id=${CLIENT_ID}` +
+    `&redirect_uri=${REDIRECT_URI}` +
+    `&scope=${scopes.join("+")}` +
+    `&state=${state}`;
+
+  res.status(302).redirect(redirect_url);
+});
+
+app.get("/callback", async (req, res) => {
+  const authorization_code = req.query.code;
+  const state = req.query.state;
+
+  let application = getState(state);
+  if (!application) {
+    return res.status(400).send({
+      message: `Your Application did not successfully install due to no application in memory`,
+    });
+  }
+
+  const payload = {
+    grant_type: "authorization_code",
+    code: authorization_code,
+    redirect_uri: REDIRECT_URI,
+    client_id: CLIENT_ID,
+    client_secret: CLIENT_SECRET,
+  };
+  // your application makes a request to exchange an authorization code for an access_token
+  const response = await axios
+    .post(ACCESS_TOKEN_ENDPOINT, qs.stringify(payload), {
+      headers: {
+        "Content-Type": "application/x-www-form-urlencoded",
+      },
+    })
```

Comment:
```suggestion
  const response = await axios
    .post(ACCESS_TOKEN_ENDPOINT, payload)
```

IIRC, if you pass axios a JSON object for the payload it figures out the right headers itself. If our endpoint accepts JSON, it's definitely more idiomatic to call it like this than to form-encode the payload

Reply:
I think we're trying to deprecate the json endpoint so we probably don't want to advertise it.

Reply:
^